### PR TITLE
Extract methods

### PR DESCRIPTION
**Pull request by mntj**
This change extracts methods from several methods to reduce code complexity, code duplication, and to improve readability.

### CODE REVIEW

**Comment by mntj** [comment]
The build has completed and the tests passed. Not sure why it's taking so long to update on GitHub.

File: lib/artoo/robot.rb [hunk 1]
```diff
@@ -186,23 +186,29 @@ def respond_to_missing?(method_name, include_private = false)
...
+    end
+
+    def initialize_device(params, type, devices)
+      type = initialize_object(Device, params, type)
+      instance_eval("def #{type.name}; return devices[:#{type.name}]; end")
+      @devices[type.name.intern] = type
+    end
+
+    def initialize_object(klass, params, type)
+      Logger.info "Initializing device #{type[:name].to_s}..."
```
Comment (deadprogram) [inline lib/artoo/robot.rb]: Now that this is being used to initialize both connections and devices, it should probably be something like `Logger.info "Initializing #{klass.to_s} #{type[:name].to_s}..."`

Comment (mntj) [inline lib/artoo/robot.rb]: Good catch, thanks for pointing that out. I'll update it

**Comment by mntj** [comment]
@deadprogram should be good to go now, thanks

File: lib/artoo/adaptors/io/digital_pin.rb [hunk 2]
```diff
@@ -50,14 +50,17 @@ def set_mode(mode)
...
+            set_pin('w')
           elsif mode =='r'
-            File.open("#{ GPIO_PATH }/gpio#{ pin_num }/direction", "w") { |f| f.write(GPIO_DIRECTION_READ) }
-            @pin_file = File.open("#{ GPIO_PATH }/gpio#{pin_num}/value", "r")
+            set_pin('r')
           end
         end
 
+        def set_pin(mode)
+          File.open("#{ GPIO_PATH }/gpio#{ pin_num }/direction", "w") { |f| f.write(GPIO_DIRECTION_WRITE) }
```
Comment (deadprogram) [inline lib/artoo/adaptors/io/digital_pin.rb]: Shouldn't the data being written here be either `GPIO_DIRECTION_WRITE` or `GPIO_DIRECTION_READ` depending on direction?

Comment (mntj) [inline lib/artoo/adaptors/io/digital_pin.rb]: Yes, apologies for overlooking this - will fix.

**Comment by mntj** [comment]
@deadprogram All good now?

**Comment by deadprogram** [comment]
Looks good, thank you!

**Pull request merged**